lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -3156,10 +3229,66 @@ static LogicalResult foldHiddenReset(RegOp reg, PatternRewriter &rewriter) {
   return success();
 }
 
+/// This canonicalizer provides the following patterns:
+///   connect(reg, and(reg, x)) ==> reg -> 0
+///   connect(reg, and(x, reg)) ==> reg -> 0
+///   connect(reg, or(reg, x))  ==> reg -> 1
+///   connect(reg, or(x, reg))  ==> reg -> 1
+///
+/// Justification: The initial value of a register is indeterminant, which means
+/// We are free to choose any initial value when optimizing the circuit. For the
+/// AND patterns, if we assume the initial value is zero, then the register will
+/// always be zero. For the OR patterns, if we assume the initial value is one,
+/// then the register will always be one.
```

Comment:
Great for including this on both rewrite patterns! Providing this as a comment makes it clear not only what the pattern is doing, but _why_ and with what justification in the spec we are using. 💯